lib/puppet/util/postgresql_validator.rb

```diff
@@ -0,0 +1,64 @@
+module Puppet
+  module Util
+    class PostgresqlValidator
+      attr_reader :resource
+
+      def initialize(resource)
+        @resource = resource
+      end
+
+      def build_psql_cmd
+        final_cmd = []
+
+        cmd_init = "#{@resource[:psql_path]} --tuples-only --quiet "
+
+        final_cmd.push cmd_init
+
+        cmd_parts = {
+          :host => "-h #{@resource[:host]}",
+          :port => "-p #{@resource[:port]}",
+          :db_username => "-U #{@resource[:db_username]}",
```

Comment:
Some short and some long arguments don't please my eyes.  I prefer `--xxx=yyy` format on scripts when possible to avoid argument parsing issues.